scripts/clean_release_notes.py

```diff
@@ -0,0 +1,192 @@
+                # Assign each commit to a category
+                for commit_line in commit_lines:
+                    category = get_commit_category(commit_line, pr_number_to_label)
+                    if category is not None:
+                        commits_by_category[category].append(commit_line)
+                # Write all commits to the output file by category
+                for category, commits in commits_by_category.items():
+                    out_f.write("## %s\n\n" % category)
+                    for commit_line in commits:
+                        out_f.write(format_commit(commit_line))
+                    out_f.write("\n")
+            else:
+                # Not a commit, just copy to the output file
+                out_f.write(line)
+    print("Wrote to %s." % output_file)
+
+
+def parse_pr_number(commit_line: str) -> int:
+    """
+    Helper function to parse PR number from commit line.
+    """
+    return int(re.match(".*pytorch/ao/pull/(.*)", commit_line).groups()[0])
+
+
+def fetch_pr_labels(commit_lines: List[str]) -> Dict[int, str]:
+    """
+    Fetch the relevant github labels starting with "topic: " from all PRs.
+    If such a label exists for a given PR, store the first one.
+    """
+    pr_number_to_label = {}
+    all_pr_numbers = [parse_pr_number(line) for line in commit_lines]
+    smallest_pr_number = min(all_pr_numbers)
+    repo = Github().get_repo("pytorch/ao")
+
+    # This call fetches 30 PRs at a time in descending order of when the PR was created
+    pulls = repo.get_pulls(state="closed")
+    for pr in pulls:
+        if pr.number < smallest_pr_number:
+            break
+        labels = [l.name for l in pr.labels if l.name.startswith("topic: ")]
+        if len(labels) > 0:
+            if VERBOSE:
+                print("Found label for PR %s: '%s'" % (pr.number, labels[0]))
+            pr_number_to_label[pr.number] = labels[0]
+    return pr_number_to_label
+
+
+def get_commit_category(commit_line: str, pr_number_to_label: Dict[int, str]) -> Optional[str]:
+    """
+    Assign the commit to a category based on:
+      (1) The github label if it exists
+      (2) Keywords in the PR title
+
+    If the commit is not meant to be user facing, remove None.
+    Otherwise, return "Improvement" by default.
+    """
+    pr_number = parse_pr_number(commit_line)
+    if pr_number in pr_number_to_label:
+        label = pr_number_to_label[pr_number]
+        if label == "topic: not user facing":
+            return None
+        if label in GITHUB_LABEL_TO_CATEGORY:
+            return GITHUB_LABEL_TO_CATEGORY[label]
+    elif any(x in commit_line.lower() for x in ["revert", "version.txt"]):
+        return None
+    elif any(x in commit_line.lower() for x in ["doc", "readme", "tutorial", "typo", "example", "spelling"]):
+        return "Documentation"
+    elif any(x in commit_line.lower() for x in ["test", "lint", " ci", "nightl"]):
```

Comment:
```suggestion
    elif any(x in commit_line.lower() for x in ["test", "lint", " ci", "nightly"]):
```

Reply:
This is on purpose actually :), so we match both nightly and nightlies (this appeared in some commit titles)